src/coreclr/vm/peimagelayout.cpp

```diff
@@ -411,34 +417,37 @@ ConvertedImageLayout::ConvertedImageLayout(PEImageLayout* source, BOOL isInBundl
         EEFileLoadException::Throw(GetPath(), COR_E_BADIMAGEFORMAT);
     LOG((LF_LOADER, LL_INFO100, "PEImage: Opening manually mapped stream\n"));
 
-#if !defined(CROSSGEN_COMPILE) && !defined(TARGET_UNIX)
-    // on Windows we may want to enable execution if the image contains R2R sections
-    // so must ensure the mapping is compatible with that
-    m_FileMap.Assign(WszCreateFileMapping(INVALID_HANDLE_VALUE, NULL,
-        PAGE_EXECUTE_READWRITE, 0,
-        source->GetVirtualSize(), NULL));
+    DWORD mapAccess;
+    DWORD viewAccess;
+    if (isInBundle && (HasNativeHeader() || HasReadyToRunHeader()))
+    {
+        // in bundle we may want to enable execution if the image contains R2R sections
+        // so must ensure the mapping is compatible with that
+        mapAccess = PAGE_EXECUTE_READWRITE;
+        viewAccess = FILE_MAP_EXECUTE | FILE_MAP_WRITE;
```

Comment:
This will be problematic with the W^X works which @janvorli  is doing. The goal of that work is to not have any RWX pages in the process... we may need to use some double-mapping tricks to avoid this here as well.

Reply:
This is what we currently do. We copy into RWX and then make it RX.

I assume W^X solves exactly this scenario and will provide a common solution/pattern. Perhaps we will be able to just remap without copying.

Reply:
I will take care of this as part of the W^X change. As @VSadov says, we already use this way in the runtime, so it will need to change.

Reply:
Perfect, just wanted to make sure we will not miss this (or make it worse).